test/common/crypto/utility_test.cc

```diff
@@ -53,48 +54,91 @@ TEST(UtilityTest, TestSha256HmacWithEmptyArguments) {
-             "183f550dac19abcf1145a7f9ced987db680e4a229cac75dee347ec9ebce1fc3dbbbb0203010001";
-
-  Common::Crypto::CryptoObjectPtr crypto_ptr(
-      Common::Crypto::UtilitySingleton::get().importPublicKey(Hex::decode(key)));
-  auto wrapper = Common::Crypto::Access::getTyped<Common::Crypto::PublicKeyObject>(*crypto_ptr);
-  EVP_PKEY* pkey = wrapper->getEVP_PKEY();
-  EXPECT_NE(nullptr, pkey);
+  // Test DER format (backward compatibility)
+  auto der_key =
+      "30820122300d06092a864886f70d01010105000382010f003082010a0282010100a7471266d01d160308d"
+      "73409c06f2e8d35c531c458d3e480e9f3191847d062ec5ccff7bc51e949d5f2c3540c189a4eca1e8633a6"
+      "2cf2d0923101c27e38013e71de9ae91a704849bff7fbe2ce5bf4bd666fd9731102a53193fe5a9a5a50644"
+      "ff8b1183fa897646598caad22a37f9544510836372b44c58c98586fb7144629cd8c9479592d996d32ff6d"
+      "395c0b8442ec5aa1ef8051529ea0e375883cefc72c04e360b4ef8f5760650589ca814918f678eee39b884"
+      "d5af8136a9630a6cc0cde157dc8e00f39540628d5f335b2c36c54c7c8bc3738a6b21acff815405afa28e5"
+      "183f550dac19abcf1145a7f9ced987db680e4a229cac75dee347ec9ebce1fc3dbbbb0203010001";
 
-  key = "badkey";
-  crypto_ptr = Common::Crypto::UtilitySingleton::get().importPublicKey(Hex::decode(key));
-  wrapper = Common::Crypto::Access::getTyped<Common::Crypto::PublicKeyObject>(*crypto_ptr);
-  pkey = wrapper->getEVP_PKEY();
-  EXPECT_EQ(nullptr, pkey);
+  Common::Crypto::PKeyObjectPtr der_crypto_ptr(
+      Common::Crypto::UtilitySingleton::get().importPublicKeyDER(Hex::decode(der_key)));
+  EVP_PKEY* der_pkey = der_crypto_ptr->getEVP_PKEY();
+  EXPECT_NE(nullptr, der_pkey) << "DER public key import failed";
+
+  // Test PEM format (new feature) - same key material as DER
+  std::string pem_key = "-----BEGIN PUBLIC KEY-----\n"
+                        "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAp0cSZtAdFgMI1zQJwG8u\n"
+                        "jTXFMcRY0+SA6fMZGEfQYuxcz/e8UelJ1fLDVAwYmk7KHoYzpizy0JIxAcJ+OAE+\n"
+                        "cd6a6RpwSEm/9/vizlv0vWZv2XMRAqUxk/5amlpQZE/4sRg/qJdkZZjKrSKjf5VE\n"
+                        "UQg2NytExYyYWG+3FEYpzYyUeVktmW0y/205XAuEQuxaoe+AUVKeoON1iDzvxywE\n"
+                        "42C0749XYGUFicqBSRj2eO7jm4hNWvgTapYwpswM3hV9yOAPOVQGKNXzNbLDbFTH\n"
+                        "yLw3OKayGs/4FUBa+ijlGD9VDawZq88RRaf5ztmH22gOSiKcrHXe40fsnrzh/D27\n"
+                        "uwIDAQAB\n"
+                        "-----END PUBLIC KEY-----";
+
+  Common::Crypto::PKeyObjectPtr pem_crypto_ptr(
+      Common::Crypto::UtilitySingleton::get().importPublicKeyPEM(pem_key));
+  EVP_PKEY* pem_pkey = pem_crypto_ptr->getEVP_PKEY();
+  EXPECT_NE(nullptr, pem_pkey) << "PEM public key import failed";
+
+  // Verify both keys are functionally equivalent
+  EXPECT_EQ(EVP_PKEY_bits(der_pkey), EVP_PKEY_bits(pem_pkey))
+      << "DER and PEM keys should have same bit size";
+  EXPECT_EQ(EVP_PKEY_size(der_pkey), EVP_PKEY_size(pem_pkey))
+      << "DER and PEM keys should have same byte size";
```

Comment:
these two are checking the same thing, just in different units. Remove one of them.